fusillade/clouddirectory.py

```diff
@@ -0,0 +1,788 @@
+            self.delete_object(obj_ref)
+
+    def delete_policy(self, policy_ref):
+        self.batch_write([self.batch_detach_policy(policy_ref, obj_ref)
+                          for obj_ref in self.list_policy_attachments(policy_ref)])
+        self.batch_write([self.batch_detach_object(parent_ref, link_name)
+                          for parent_ref, link_name in self.list_object_parents(policy_ref)])
+        ad.delete_object(DirectoryArn=self._dir_arn, ObjectReference={'Selector': policy_ref})
+
+    def delete_object(self, obj_ref):
+        self.batch_write([self.batch_detach_policy(policy_ref, obj_ref)
+                          for policy_ref in self.list_object_policies(obj_ref)])
+        self.batch_write([self.batch_detach_object(parent_ref, link_name)
+                          for parent_ref, link_name in self.list_object_parents(obj_ref)])
+        ad.delete_object(DirectoryArn=self._dir_arn, ObjectReference={'Selector': obj_ref})
+
+    @staticmethod
+    def batch_detach_policy(policy_ref, object_ref):
+        return {
+            'DetachPolicy': {
+                'PolicyReference': {'Selector': policy_ref},
+                'ObjectReference': {'Selector': object_ref}
+            }
+        }
+
+    def batch_create_object(self, parent, name, facet_name, object_attribute_list):
+        """
+        batch write operation
+        :param parent:
+        :param name:
+        :param facet_name:
+        :param object_attribute_list:
+        :return:
+        """
+        return {'CreateObject': {
+            'SchemaFacet': [
+                {
+                    'SchemaArn': self._schema,
+                    'FacetName': facet_name
+                },
+            ],
+            'ObjectAttributeList': object_attribute_list,
+            'ParentReference': {
+                'Selector': parent
+            },
+            'LinkName': name,
+        }
+        }
+
+    def batch_get_attributes(self, obj_ref, facet, attributes: typing.List[str]):
+        return {
+            'GetObjectAttributes': {
+                'ObjectReference': {
+                    'Selector': obj_ref
+                },
+                'SchemaFacet': {
+                    'SchemaArn': self._schema,
+                    'FacetName': facet
+                },
+                'AttributeNames': attributes
+            }
+        }
+
+    @staticmethod
+    def batch_attach_object(parent, child, name):
+        """
+        batch write operation
+        :param parent:
+        :param child:
+        :param name:
+        :return:
+        """
+        return {
+            'AttachObject': {
+                'ParentReference': {
+                    'Selector': parent
+                },
+                'ChildReference': {
+                    'Selector': child
+                },
+                'LinkName': name
+            }
+        }
+
+    @staticmethod
+    def batch_detach_object(parent, link_name):
```

Comment:
Many of these could use docstrings.  If there's a reference doc, I think a link might suffice.